src/test/java/ua/chernonog/onlinebookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,113 @@
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.CATEGORIES_BASE_URL;
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.CATEGORY_BY_ID_URL;
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.CATEGORY_DESCRIPTION;
+import static ua.chernonog.onlinebookstore.config.util.CategoryConstants.CATEGORY_NAME;
+import static ua.chernonog.onlinebookstore.config.util.UtilConstants.JSON_PATH_FIRST_ITEM_EXISTS;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import ua.chernonog.onlinebookstore.dto.request.category.CreateCategoryDto;
+import ua.chernonog.onlinebookstore.dto.response.book.BookDtoWithoutCategoryIds;
+import ua.chernonog.onlinebookstore.dto.response.category.CategoryResponseDto;
+import ua.chernonog.onlinebookstore.service.CategoryService;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
+    private static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Mock
+    private CategoryService categoryService;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext context) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(context)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @WithMockUser(roles = "ADMIN")
+    @DisplayName("Save a new category by admin")
+    void save_WithValidCategory_ShouldReturnSavedCategory() throws Exception {
+        CreateCategoryDto createCategoryDto = new CreateCategoryDto();
+        createCategoryDto.setName(CATEGORY_NAME);
+        createCategoryDto.setDescription(CATEGORY_DESCRIPTION);
+
+        CategoryResponseDto responseDto = new CategoryResponseDto();
+        responseDto.setId(1L);
+        responseDto.setName(createCategoryDto.getName());
+        responseDto.setDescription(createCategoryDto.getDescription());
+
+        when(categoryService.save(any(CreateCategoryDto.class))).thenReturn(responseDto);
+
+        mockMvc.perform(post(CATEGORIES_BASE_URL)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(createCategoryDto))
+                        .with(csrf()))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name").value("Science Fiction"));
```

Comment:
Why sometimes you use constants for jsonPath and sometimes not? Please be consistent and choose approach